### PR TITLE
Use official client library in Telegram bot

### PR DESCRIPTION
**Pull request by antonyshchenko**
I've abstracted the Telegram client into a simple interface so that bot implementation does not depend on a specific library

### CODE REVIEW

File: telegrambot/src/main/kotlin/co/chainring/telegrambot/app/BotTelegramClient.kt [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+import org.telegram.telegrambots.meta.api.methods.send.SendMessage
+import org.telegram.telegrambots.meta.api.methods.updatingmessages.DeleteMessage
+import org.telegram.telegrambots.meta.api.objects.Update
+import org.telegram.telegrambots.meta.api.objects.replykeyboard.ForceReplyKeyboard
+import org.telegram.telegrambots.meta.api.objects.replykeyboard.InlineKeyboardMarkup
+import org.telegram.telegrambots.meta.api.objects.replykeyboard.buttons.InlineKeyboardButton
+import org.telegram.telegrambots.meta.api.objects.replykeyboard.buttons.InlineKeyboardRow
+
+class BotTelegramClient(private val token: String) : Bot.TelegramClient {
+    private val app = TelegramBotsLongPollingApplication()
```
Comment (bflood366) [inline telegrambot/src/main/kotlin/co/chainring/telegrambot/app/BotTelegramClient.kt:17]: not sure if its needed but is there any wire level logging to aid in debugging or is there anything we would want to log

File: telegrambot/src/main/kotlin/co/chainring/telegrambot/app/BotTelegramClient.kt [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        app.registerBot(
+            token,
+            object : LongPollingSingleThreadUpdateConsumer {
+                override fun consume(update: Update) {
+                    val message = update.message
+                    val callbackQuery = update.callbackQuery
+
+                    if (message != null && message.hasText()) {
+                        if (message.text == "/start") {
+                            inputHandler(
```
Comment (bflood366) [inline telegrambot/src/main/kotlin/co/chainring/telegrambot/app/BotTelegramClient.kt:30]: do we know what would happen if the inputHandler threw an unexpected exception. Would it cause this consume thread to exist or does `LongPollingSingleThreadUpdateConsumer` catch it and keep going and if that did happen would we see anything logged anywhere

**Review by bflood366 — APPROVED** [review]
looks great 👍

**Review by imykolenko — APPROVED** [review]
nice!

File: telegrambot/src/main/kotlin/co/chainring/telegrambot/app/BotTelegramClient.kt (hunk 1 above)
Comment (antonyshchenko) [reply telegrambot/src/main/kotlin/co/chainring/telegrambot/app/BotTelegramClient.kt:18]: Added debug logging for input and output messages, although we should turn it off on prod in order not to leak private keys

File: telegrambot/src/main/kotlin/co/chainring/telegrambot/app/BotTelegramClient.kt (hunk 2 above)
Comment (antonyshchenko) [reply telegrambot/src/main/kotlin/co/chainring/telegrambot/app/BotTelegramClient.kt]: I've checked and it does not exit or crash on unhandled exceptions. However I've wrapped handling in try/catch anyways in order to log the error and send `Something went wrong, please try again later` back to the user.

**Pull request merged**